## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 411069`
Commit: `0b3e4f1dd3d3a65f58a3d79cb08f69090ccc97dc`

---
### `x86_64-linux`
<details>
  <summary>:x: 10 packages failed to build:</summary>
  <ul>
    <li>home-assistant-component-tests.apple_tv</li>
    <li>ja2-stracciatella</li>
    <li>python312Packages.miniaudio</li>
    <li>python312Packages.miniaudio.dist</li>
    <li>python312Packages.pyatv</li>
    <li>python312Packages.pyatv.dist</li>
    <li>python313Packages.miniaudio</li>
    <li>python313Packages.miniaudio.dist</li>
    <li>python313Packages.pyatv</li>
    <li>python313Packages.pyatv.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>miniaudio</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>ja2-stracciatella</summary>
<pre>-- Installing: /build/source/build/lib-string_theory/include/string_theory/format
-- Installing: /build/source/build/lib-string_theory/include/string_theory/iostream
-- Installing: /build/source/build/lib-string_theory/include/string_theory/stdio
-- Installing: /build/source/build/lib-string_theory/include/string_theory/string
-- Installing: /build/source/build/lib-string_theory/include/string_theory/string_stream
-- Installing: /build/source/build/lib-string_theory/include/string_theory/utf_conversion
-- Installing: /build/source/build/lib-string_theory/lib/cmake/string_theory/string_theory-config.cmake
-- Installing: /build/source/build/lib-string_theory/lib/cmake/string_theory/string_theory-config-version.cmake
-- Installing: /build/source/build/lib-string_theory/lib/cmake/string_theory/string_theory-targets.cmake
[100%] Completed 'string_theory'
[100%] Built target string_theory
-- </string_theory>
-- Using system miniaudio
-- Looking for C++ include miniaudio.h
-- Looking for C++ include miniaudio.h - not found
CMake Error at dependencies/lib-miniaudio/CMakeLists.txt:16 (message):
  miniaudio.h not found


-- Configuring incomplete, errors occurred!</pre>
</details>
<details>
<summary>python312Packages.miniaudio</summary>
<pre>!!
  self._finalize_license_expression()
running bdist_wheel
running build
running build_py
creating build/lib.linux-x86_64-cpython-312
copying miniaudio.py -> build/lib.linux-x86_64-cpython-312
running build_ext
generating cffi module 'build/temp.linux-x86_64-cpython-312/_miniaudio.c'
creating build/temp.linux-x86_64-cpython-312
building '_miniaudio' extension
creating build/temp.linux-x86_64-cpython-312/build/temp.linux-x86_64-cpython-312
gcc -fno-strict-overflow -Wsign-compare -DNDEBUG -g -O3 -Wall -I/nix/store/qizipyz9y17nr4w4gmxvwd3x4k0bp2rh-libxcrypt-4.4.38/include -fPIC -DMA_NO_GENERATION=1 -DMA_NO_ENCODING=1 -DMA_NO_RESOURCE_MANAGER=1 -DMA_NO_NODE_GRAPH=1 -DMA_NO_ENGINE=1 -I/build/source -I/build/source -I/nix/store/8w718rm43x7z73xhw9d6vh8s4snrq67h-python3-3.12.10/include/python3.12 -c build/temp.linux-x86_64-cpython-312/_miniaudio.c -o build/temp.linux-x86_64-cpython-312/build/temp.linux-x86_64-cpython-312/_miniaudio.o -g1 -O3 -ffast-math
build/temp.linux-x86_64-cpython-312/_miniaudio.c:579:14: fatal error: miniaudio/extras/stb_vorbis.c: No such file or directory
  579 |     #include "miniaudio/extras/stb_vorbis.c"
      |              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
error: command '/nix/store/0fsnicvfpf55nkza12cjnad0w84d6ba7-gcc-wrapper-14.2.1.20250322/bin/gcc' failed with exit code 1

ERROR Backend subprocess exited when trying to invoke build_wheel</pre>
</details>
<details>
<summary>python313Packages.miniaudio</summary>
<pre>!!
  self._finalize_license_expression()
running bdist_wheel
running build
running build_py
creating build/lib.linux-x86_64-cpython-313
copying miniaudio.py -> build/lib.linux-x86_64-cpython-313
running build_ext
generating cffi module 'build/temp.linux-x86_64-cpython-313/_miniaudio.c'
creating build/temp.linux-x86_64-cpython-313
building '_miniaudio' extension
creating build/temp.linux-x86_64-cpython-313/build/temp.linux-x86_64-cpython-313
gcc -fno-strict-overflow -Wsign-compare -DNDEBUG -g -O3 -Wall -I/nix/store/qizipyz9y17nr4w4gmxvwd3x4k0bp2rh-libxcrypt-4.4.38/include -fPIC -DMA_NO_GENERATION=1 -DMA_NO_ENCODING=1 -DMA_NO_RESOURCE_MANAGER=1 -DMA_NO_NODE_GRAPH=1 -DMA_NO_ENGINE=1 -I/build/source -I/build/source -I/nix/store/2mab9iiwhcqwk75qwvp3zv0bvbiaq6cs-python3-3.13.3/include/python3.13 -c build/temp.linux-x86_64-cpython-313/_miniaudio.c -o build/temp.linux-x86_64-cpython-313/build/temp.linux-x86_64-cpython-313/_miniaudio.o -g1 -O3 -ffast-math
build/temp.linux-x86_64-cpython-313/_miniaudio.c:579:14: fatal error: miniaudio/extras/stb_vorbis.c: No such file or directory
  579 |     #include "miniaudio/extras/stb_vorbis.c"
      |              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
error: command '/nix/store/0fsnicvfpf55nkza12cjnad0w84d6ba7-gcc-wrapper-14.2.1.20250322/bin/gcc' failed with exit code 1

ERROR Backend subprocess exited when trying to invoke build_wheel</pre>
</details>
</details>

--- a/pkgs/by-name/ja/ja2-stracciatella/package.nix
+++ b/pkgs/by-name/ja/ja2-stracciatella/package.nix
@@ -101,6 +101,8 @@ stdenv.mkDerivation rec {
   '';
 
   meta = {
+    # Fails to build on x86_64-linux as of 2025-03-16 and potentially earlier
+    broken = true;
     description = "Jagged Alliance 2, with community fixes";
     license = "SFI Source Code license agreement";
     homepage = "https://ja2-stracciatella.github.io/";

--- a/pkgs/by-name/mi/miniaudio/package.nix
+++ b/pkgs/by-name/mi/miniaudio/package.nix
@@ -2,32 +2,52 @@
   lib,
   stdenv,
   fetchFromGitHub,
+  testers,
 }:
-stdenv.mkDerivation rec {
+stdenv.mkDerivation (finalAttrs: {
   pname = "miniaudio";
   version = "0.11.22";
 
   src = fetchFromGitHub {
     owner = "mackron";
     repo = "miniaudio";
-    rev = version;
+    rev = finalAttrs.version;
     hash = "sha256-o/7sfBcrhyXEakccOAogQqm8dO4Szj1QSpaIHg6OSt4=";
   };
 
-  installPhase = ''
-    mkdir -p $out
-    cp -r $src/* $out/
+  postInstall = ''
+    mkdir -p $out/include
+    mkdir -p $out/lib/pkgconfig
+
+    cp $src/miniaudio.h $out/include
+    ln -s $out/include/miniaudio.h $out
+
+    cp -r $src/extras $out/
+
+    cat <<EOF >$out/lib/pkgconfig/miniaudio.pc
+    prefix=$out
+    includedir=$out/include
+
+    Name: miniaudio
+    Description: An audio playback and capture library in a single source file.
+    Version: $version
+    Cflags: -I$out/include
+    Libs: -lm -lpthread -latomic
+    EOF
   '';
+
+  passthru.tests.pkg-config = testers.testMetaPkgConfig finalAttrs.finalPackage;
 
   meta = with lib; {
     description = "Single header audio playback and capture library written in C";
     homepage = "https://github.com/mackron/miniaudio";
-    changelog = "https://github.com/mackron/miniaudio/blob/${src.rev}/CHANGES.md";
+    changelog = "https://github.com/mackron/miniaudio/blob/${finalAttrs.version}/CHANGES.md";
     license = with licenses; [
       unlicense # or
       mit0
     ];
     maintainers = [ maintainers.jansol ];
+    pkgConfigModules = [ "miniaudio" ];
     platforms = platforms.all;
   };
-}
+})

